bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -277,6 +292,44 @@ private Map<String, Object> getRulesProperties() {
         return customRulesProperties;
     }
 
+    /**
+     * Returns {@code Map<String, String>} containing the key value pairs associated with this resource
+     */
+    @JsonIgnore
+    public Map<String, String> getHeadlessCustomProperties() {
+        Set<String> groupNames = new LinkedHashSet<>();
```

Comment:
Why it needs to be a LinkedHashSet as compared to a normal HashSet? I fail to notice use of ordered insertion here.